.github/workflows/ci-cd-pipeline.yml

```diff
@@ -1,88 +1,74 @@
 name: CI-CD Pipeline
 
 on:
+  push:
+    branches: [test-branch]
   pull_request:
-    branches:
-      - main
+    branches: [main]
   schedule:
-    - cron: "0 0 * * *"  # Запуск по расписанию (каждый день в 00:00)
+    - cron: "0 0 * * *"
   workflow_dispatch:
     inputs:
       browser:
         description: 'Select browser for UI tests'
         required: true
         default: 'chrome'
         type: choice
-        options:
-          - chrome
-          - firefox
-          - edge
+        options: [chrome, firefox, edge]
 
 jobs:
   api-tests:
     runs-on: ubuntu-latest
-    name: Run API Tests
+    name: API Tests
     steps:
-      - name: Checkout API repository
-        uses: actions/checkout@v2
+      - name: Checkout API repo
+        uses: actions/checkout@v4
         with:
-          repository: 'AlibekNamazbayev/APIAutomation'
+          repository: AlibekNamazbayev/APIAutomation
           token: ${{ secrets.GITHUB_TOKEN }}
-        
-      - name: Setup .NET
-        uses: actions/setup-dotnet@v3
+          path: api-repo
+
+      - name: Setup .NET 9
+        uses: actions/setup-dotnet@v4
         with:
-          dotnet-version: '9.0.x'
+          dotnet-version: | 
+            9.0.x
+            8.0.x  
 
       - name: Restore dependencies
-        run: dotnet restore APIAutomation.sln
-
-      - name: Build API project
-        run: dotnet build APIAutomation.sln --configuration Release
-
-      - name: Run API tests
-        run: dotnet test APITest.Tests/ApiTest.Tests.csproj --logger "trx"
-
-      - name: Upload API test results
-        uses: actions/upload-artifact@v2
-        with:
-          name: api-test-results
-          path: **/*.trx
+        run: dotnet restore ApiTestSolution.sln
+        working-directory: ./api-repo
 
   ui-tests:
     runs-on: ubuntu-latest
-    name: Run UI Tests
+    name: UI Tests
     needs: api-tests
     steps:
-      - name: Checkout UI repository
-        uses: actions/checkout@v2
+      - name: Checkout UI repo
+        uses: actions/checkout@v4
         with:
-          repository: 'AlibekNamazbayev/.NET-Web-UI-Test-Automation-Tools-Part-2'
+          repository: AlibekNamazbayev/.NET-Web-UI-Test-Automation-Tools-Part-2
           token: ${{ secrets.GITHUB_TOKEN }}
+          path: ui-repo
 
       - name: Setup .NET
-        uses: actions/setup-dotnet@v3
+        uses: actions/setup-dotnet@v4
         with:
-          dotnet-version: '9.0.x'
+          dotnet-version: | 
```

Comment:
why do u need to install dotnet 2 times? this is one host machine executing the tests. One time is enough